src/tests/CommandParserTests.cpp

```diff
@@ -7,318 +7,261 @@
+  std::string const EmptyNexusStructure;
+  auto MessageBuffer = buildRunStartMessage(
+      InstrumentNameInput, RunNameInput, EmptyNexusStructure, JobIDInput,
+      ServiceIDInput, BrokerInput, FilenameInput, StartTimeInput,
+      StopTimeInput);
+
+  ASSERT_THROW(
+      FileWriter::CommandParser::extractStartInformation(MessageBuffer),
+      std::runtime_error);
 }
 
 TEST(CommandParserSadStartTests, IfNoBrokerThenThrows) {
-  std::string Command(R"""(
-{
-  "cmd": "FileWriter_new",
-  "job_id": "qw3rty",
-  "file_attributes": {
-    "file_name": "a-dummy-name-01.h5"
-  },
-  "nexus_structure": { }
-})""");
-
-  ASSERT_THROW(FileWriter::CommandParser::extractStartInformation(
-                   nlohmann::json::parse(Command)),
-               std::runtime_error);
+  std::string const EmptyBroker;
+  auto MessageBuffer = buildRunStartMessage(
+      InstrumentNameInput, RunNameInput, NexusStructureInput, JobIDInput,
+      ServiceIDInput, EmptyBroker, FilenameInput, StartTimeInput,
+      StopTimeInput);
+
+  ASSERT_THROW(
+      FileWriter::CommandParser::extractStartInformation(MessageBuffer),
+      std::runtime_error);
 }
 
 TEST(CommandParserSadStartTests, IfBrokerIsWrongFormThenThrows) {
-  std::string Command(R"""(
-{
-  "cmd": "FileWriter_new",
-  "job_id": "qw3rty",
-  "file_attributes": {
-    "file_name": "a-dummy-name-01.h5"
-  },
-  "broker": "1234:somehost",
-  "nexus_structure": { }
-})""");
-
-  ASSERT_THROW(FileWriter::CommandParser::extractStartInformation(
-                   nlohmann::json::parse(Command)),
-               std::runtime_error);
+  std::string const BrokerInvalidFormat = "1234:somehost";
+  auto MessageBuffer = buildRunStartMessage(
+      InstrumentNameInput, RunNameInput, NexusStructureInput, JobIDInput,
+      ServiceIDInput, BrokerInvalidFormat, FilenameInput, StartTimeInput,
+      StopTimeInput);
+
+  ASSERT_THROW(
+      FileWriter::CommandParser::extractStartInformation(MessageBuffer),
+      std::runtime_error);
 }
 
 TEST(CommandParserStartTests, IfNoStartTimeThenUsesSuppliedCurrentTime) {
-  std::string Command(R"""(
-{
-  "cmd": "FileWriter_new",
-  "job_id": "qw3rty",
-  "broker": "localhost:9092",
-  "file_attributes": {
-    "file_name": "a-dummy-name-01.h5"
-  },
-  "nexus_structure": { }
-})""");
+  // Start time from flatbuffer is 0 if not supplied when message constructed
+  uint64_t const NoStartTime = 0;
+  auto MessageBuffer = buildRunStartMessage(
+      InstrumentNameInput, RunNameInput, NexusStructureInput, JobIDInput,
+      ServiceIDInput, BrokerInput, FilenameInput, NoStartTime, StopTimeInput);
 
   auto FakeCurrentTime = std::chrono::milliseconds{987654321};
 
   auto StartInfo = FileWriter::CommandParser::extractStartInformation(
-      nlohmann::json::parse(Command), FakeCurrentTime);
+      MessageBuffer, FakeCurrentTime);
 
   ASSERT_EQ(FakeCurrentTime, StartInfo.StartTime);
 }
 
-TEST(CommandParserStartTests, IfNoStopTimeThenSetToZero) {
```

Comment:
There is too much code for me to review in detail but I would like to have you comment on if you have replaced every unit tests that you removed with a new one or if one or more unit tests were removed entirely.

Reply:
Flatbuffers has no concept of "NoStopTime", the field is always present and therefore it will always give a value. So this test no longer made sense.

Reply:
👍